static/js/image_upload_widget.js

```diff
@@ -0,0 +1,75 @@
+"use strict";
+
+function image_upload_complete(spinner, upload_text, delete_button) {
+    spinner.css({visibility: "hidden"});
+    upload_text.show();
+    delete_button.show();
+}
+
+function image_upload_start(spinner, upload_text, delete_button) {
+    spinner.css({visibility: "visible"});
+    upload_text.hide();
+    delete_button.hide();
+}
+
+class ImageUploadWidget {
+    constructor(widget) {
+        this.widget = widget;
+        this.url = this.get_url(widget);
+    }
+
+    get_url(widget) {
+        switch (widget) {
+            case "realm-night-logo-upload-widget":
+                return "/json/realm/logo";
+            case "realm-day-logo-upload-widget":
+                return "/json/realm/logo";
+            case "realm-icon-upload-widget":
+                return "/json/realm/icon";
+        }
+    }
```

Comment:
This interface is messy; we should just have the caller pass  `url` as a second argument rather than having this message switch statement.  

And the other conditional in this file for the night logo widget should instead be looking at a parameter like `night_param=false`.

Finally, the `widget` parameter should probably be called `selector`, since after the above changes, that'd be its role in the code here.